src/App.tsx

```diff
@@ -1,61 +1,119 @@
+import React, { useState, useEffect } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+  const [userId, setUserId] = useState('');
+  const [userError, setUserError] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>(todosFromServer);
+
+  const onTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+
+    const englishRegex = /[A-Za-z]/;
+    const ukrainianRegex = /[А-ЩЬЮЯҐЄІЇа-щьюяґєії]/;
+
+    if (englishRegex.test(e.target.value)) {
+      setTitle(e.target.value.replace(/[^A-Za-z0-9\s]+/g, ""));
+    }
+
+    if (ukrainianRegex.test(e.target.value)) {
+      setTitle(e.target.value.replace(/[^А-ЩЬЮЯҐЄІЇа-щьюяґєії0-9\s]+/g, ''))
+    }
+
+    setTitleError(false);
+  };
+
+  const onSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(e.target.value);
+    setUserError(false);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (!userId) {
+      setUserError(true);
+    }
+
+    if (!title || !userId) {
+      return;
+    }
+
+    setTodos(prevState => [
+      ...prevState,
+      {
+        id: !todos.length ? Math.floor(Math.random() * (200 - 100 + 1)) + 100 : Math.max(...todos.map(todo => todo.id)) + 1,
```

Comment:
The logic for generating a new todo ID does not handle the case where the `todos` array is empty. This could result in an invalid ID. Consider adding a check or default value to handle this scenario, ensuring a valid ID is always generated.